module/builder/consensus/builder.go

```diff
@@ -450,6 +470,16 @@ func (b *Builder) getInsertableReceipts(parentID flow.Identifier) ([]*flow.Execu
 		receipts = receipts[:b.cfg.maxReceiptCount]
 	}
 
+	// sanity check: parent result should be on the fork
+	for _, receiptCandidate := range receipts {
+		// check that parent result is in fork:
+		if _, found := knownResults[receiptCandidate.ExecutionResult.PreviousResultID]; !found {
```

Comment:
how could this happen? isn't the matching engine checking it already and return incoporated results?